Automattic/wp-calypso
# Storage Add-ons: Fix add-on dropdown large currency overflow

Related to https://github.com/Automattic/wp-calypso/pull/81338#pullrequestreview-1626537670
Based off of these designs dyXNnR0oHu8H47eUES8ktJ-fi-554_23399

## Proposed Changes

* If the selected add-on price is too long ( more than 7 characters ), display the price on a line separate from and below the dropdown ( instead of within the dropdown )
* If the selected add-on price is in the comparison grid, display the price on a line separate from and below the dropdown ( instead of within the dropdown )
* When rendering storage add-on option prices that are too long, display them on a new line. If it's still too long, display  `/month` string on its own line as well

## Screenshots

### Before
<img width="1260" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1842898/f809dc82-84c2-44ed-90ca-3fc9c66a0622">

### After
#### Storage Add-on dropdown options

| Small currency values | Large currency values | Extra Large currency values |
| ----- | ----- | ----- |
| <img width="242" alt="Screenshot 2023-09-14 at 4 01 58 PM" src="https://github.com/Automattic/wp-calypso/assets/5414230/8cf7fe53-f1df-4b2e-927d-a39db04131b8"> | <img width="194" alt="Screenshot 2023-09-14 at 4 13 02 PM" src="https://github.com/Automattic/wp-calypso/assets/5414230/3ebec3e4-26aa-4e2d-8828-fc5b9ca76a50"> | <img width="199" alt="Screenshot 2023-09-14 at 4 14 45 PM" src="https://github.com/Automattic/wp-calypso/assets/5414230/0856d5e5-5fcd-4524-b72c-a354c27ec27f"> |

#### Feature Grid selected price
Add-on prices will now be displayed on a separate line if any of the prices exceed 7 characters ( including currency symbols, periods, commas, etc. ). Otherwise, display price within the dropdown

<img width="1518" alt="Screenshot 2023-09-26 at 4 40 50 PM" src="https://github.com/Automattic/wp-calypso/assets/5414230/29aae945-1fd3-4633-9d2d-fb493e5648bf">

<img width="1501" alt="Screenshot 2023-09-26 at 4 42 30 PM" src="https://github.com/Automattic/wp-calypso/assets/5414230/db8f4ba1-7a7b-44f4-9594-235aace33357">

#### Comparison Grid selected price
Prices will now always be displayed on a separate line in the comparison grid because, in the majority of cases, the column width is too narrow to support even smaller currencies

<img width="1283" alt="Screenshot 2023-09-26 at 4 20 31 PM" src="https://github.com/Automattic/wp-calypso/assets/5414230/0de61233-f61f-4f33-832f-84a3eaff9149">

#### Sticky CTAs
##### Small currency values
<img width="1451" alt="270538458-c51b0c90-9a67-4481-a32f-a4e390f2e8de" src="https://github.com/Automattic/wp-calypso/assets/5414230/10fc2798-6e3d-4b2c-9bd7-ad1e4993d456">

##### Large currency values + Extra Large Currency values
<img width="1447" alt="Screenshot 2023-09-25 at 9 35 03 PM" src="https://github.com/Automattic/wp-calypso/assets/5414230/ab430e30-0991-445f-870f-065130c713e5">

## Testing Instructions




* Check out this branch or use calypso live
* Navigate to `/start?flags=plans%2Fupgradeable-storage`
* Verify that if the add-on prices are small enough, they will render in line with the storage add-on options
* Switch to a currency with larger values, like IDR
* Verify that if the add-on prices are too large, they will render on a new line below the storage add-on options
* Smoke test a few other flows and the `/plans` page
  * /setup/newsletter ( Stepper flows should be unaffected by our changes )
      * navigate to http://calypso.localhost:3000/setup/newsletter as a logged in user
      * Complete onboarding steps until you’ve reached the plans page
  * /start/domains
      * Navigate to http://calypso.localhost:3000/start/domains
      * Select a custom domain like .org, .blog, etc.
      * Navigate to the plans page
  * /start/personal
      * Navigate to http://calypso.localhost:3000/start/personal
      * Select any domain, free or custom

## Pre-merge Checklist
